src/Translumo.TTS/Engines/SileroTTSEngine.cs

```diff
@@ -0,0 +1,144 @@
+﻿using System.Media;
+using Translumo.Infrastructure.Constants;
+using Translumo.Infrastructure.Python;
+
+namespace Translumo.TTS.Engines;
+
+public class SileroTTSEngine : ITTSEngine
+{
+    private dynamic _ipython;
+    private dynamic _model;
+    private string _voice;
+    private readonly string _modelPath;
+    private readonly PythonEngineWrapper _pythonEngine;
+    private readonly List<IDisposable> _pyObjects = new();
+    private readonly SoundPlayer _player = new SoundPlayer();
+    private CancellationTokenSource _cancellationTokenSource = new CancellationTokenSource();
+
+    private const string _sileroModelDirectory = "Silero";
+
+
+    private const int _wave_rate = 48000;
+
+
+    public SileroTTSEngine(PythonEngineWrapper pythonEngine, string langCode)
+    {
+        _pythonEngine = pythonEngine;
+        _modelPath = GetModelFullPath(langCode);
+        Init();
+    }
+
+    private void Init()
+    {
+        _pythonEngine.Init();
+
+        _pythonEngine.Execute(() =>
+        {
+            var np = _pythonEngine.Import("numpy");
+            _ipython = _pythonEngine.Import("IPython");
+            dynamic torch = _pythonEngine.Import("torch");
+
+            using dynamic device = torch.device("cpu");
+            torch.set_num_threads(4);
+
+            _model = torch.package.PackageImporter(_modelPath).load_pickle("tts_models", "model");
+            _model.to(device);
+
+            _pyObjects.Add(_model);
+            _pyObjects.Add(np);
+            _pyObjects.Add(torch);
+            _pyObjects.Add(_ipython);
+        });
+
+        _voice = ((string[])_model.speakers).First();
+    }
+
+    public void SpeechText(string text)
+    {
+        _cancellationTokenSource.Cancel();
+        _cancellationTokenSource = new CancellationTokenSource();
```

Comment:
and how it may be fixed?
I can wrap it into the lock section like 
```
lock(){
   _cancellationTokenSource.Cancel();
   _cancellationTokenSource = new CancellationTokenSource();
   var currentToken =  _cancellationTokenSource;
 }
 Task.Factory.StartNew.... <with currentToken >
 ```
 
 will be this enough?

Reply:
Yes, that will be fine, thanks

Reply:
done